pkg/plugins/backendplugin/manager.go

```diff
@@ -209,30 +210,57 @@ func (m *manager) CallResource(config PluginConfig, c *models.ReqContext, path s
 		Body:    body,
 	}
 
-	res, err := p.callResource(clonedReq.Context(), req)
+	stream, err := p.callResource(clonedReq.Context(), req)
 	if err != nil {
 		c.JsonApiErr(500, "Failed to call resource", err)
 		return
 	}
 
-	// Make sure a content type always is returned in response
-	if _, exists := res.Headers["Content-Type"]; !exists {
-		res.Headers["Content-Type"] = []string{"application/json"}
-	}
+	processedStreams := 0
 
-	for k, values := range res.Headers {
-		if k == "Set-Cookie" {
-			continue
+	for {
+		resp, err := stream.Recv()
+		if err == io.EOF {
+			if processedStreams == 0 {
+				c.JsonApiErr(500, "Received empty resource response ", nil)
+			}
+			return
+		}
+		if err != nil {
+			if processedStreams == 0 {
+				c.JsonApiErr(500, "Failed to receive response from resource call", err)
+			} else {
+				p.logger.Error("Failed to receive response from resource call", "error", err)
+			}
+			return
+		}
+
+		// Expected that headers and status are only part of first stream
+		if processedStreams == 0 {
+			// Make sure a content type always is returned in response
+			if _, exists := resp.Headers["Content-Type"]; !exists {
+				resp.Headers["Content-Type"] = []string{"application/json"}
+			}
+
+			for k, values := range resp.Headers {
+				if k == "Set-Cookie" {
```

Comment:
If it is important that we remove this header, should this be a function that is called elsewhere?

Reply:
Could probably extract a function

Reply:
But this is the only place so vote for keeping it as is for now. We should add some tests eventually, but later I'm afraid :smile: Do you agree?

Reply:
If only place then agree. Maybe just add comment about why that cookie is skipped.

Reply:
I wasn't clear. We have some places where we delete the `Set-Cookie` header, but it operates on the `http.Header` struct. Added comment.